### PR TITLE
Add test for varsub in secret

### PR DESCRIPTION
**Pull request by SomtochiAma**
Fixes: #502 
Fixes: #504

Signed-off-by: Somtochi Onyekwere <somtochionyekwere@gmail.com>

### CODE REVIEW

File: docs/spec/v1beta2/kustomization.md [hunk 1]
```diff
@@ -756,6 +756,18 @@ spec:
         name: cluster-secret-vars
 ```
 
+Note that for substituting variables in a secret, `spec.stringData` field must be used i.e
+```yaml
```
Comment (stefanprodan) [inline docs/spec/v1beta2/kustomization.md:760]: Add new line above this please.

**Review by stefanprodan — CHANGES_REQUESTED** [review]

**Review by makkes — APPROVED** [review]

**Comment by stefanprodan** [comment]
@makkes I see that you've approved this PR, but the doc formatting is wrong, it will not render on fluxcd.io due to that empty line missing.

**Comment by makkes** [comment]
> @makkes I see that you've approved this PR, but the doc formatting is wrong, it will not render on fluxcd.io due to that empty line missing.

Yeah, my usual way in these situations is to approve the PR and assume that prior change requests are properly addressed. Does that sound reasonable?

**Comment by stefanprodan** [comment]
> Does that sound reasonable?

Yeap all good

**Review by stefanprodan — APPROVED** [review]

**Pull request merged**